spvc/test/run_spirv_cross_tests.py

```diff
@@ -45,127 +43,112 @@
 test_count = 0
 pass_count = 0
 
-not_used, tmpfile = tempfile.mkstemp()
-devnull = open(os.devnull, 'w')
-
-def mkdir_p(filepath):
-    ''' make sure directory exists to contain given file '''
-    dirpath = os.path.dirname(filepath)
-    try:
-        os.makedirs(dirpath)
-    except OSError as exc:
-        if not (exc.errno == errno.EEXIST and os.path.isdir(dirpath)):
-            raise
+def check_call(cmd):
+    global devnull
+    if subprocess.call(cmd, stdout=devnull):
+        print('failed command: ', ' '.join(cmd))
+        sys.exit(1)
 
-def spirv_as(inp, out):
-    subprocess.check_call([spirv_as_path, '-o', out, inp], stdout=devnull)
+def spirv_as(inp, out, flags):
+    check_call([spirv_as_path] + flags + ['-o', out, inp])
 
-def spirv_opt(inp, out):
-    subprocess.check_call([spirv_opt_path, '-o', out, inp], stdout=devnull)
+def spirv_opt(inp, out, flags):
+    check_call([spirv_opt_path] + flags + ['--skip-validation', '-O', '-o', out, inp])
 
-def glslangValidator(inp, out):
-    subprocess.check_call([glslangValidator_path , '--target-env', 'vulkan1.1', '-V', '-o',
-                          out, inp], stdout=devnull)
+def glslang_compile(inp, out, flags):
+    check_call([glslangValidator_path] + flags + ['-o', out, inp])
 
 def spvc(inp, out, flags):
-    return subprocess.call([spvc_path, '-o', out, '--validate=vulkan1.1'] + flags + [inp],
-                           stdout=devnull)
-
-def compile_shader(source_path, binary_path, filename, optimize):
-    # Note: filename is the last component of source_path.
-    mkdir_p(binary_path)
-    if '.asm.' in filename:
-        spirv_as(source_path, binary_path)
-    else:
-        glslangValidator(source_path, binary_path)
-    if optimize:
-        spirv_opt(binary_path, binary_path)
-
-def do_test(binary_path, reference_path, flags):
-    global test_count, pass_count
+    global devnull
+    global test_count
     test_count += 1
```

Comment:
I would not modify the test counter inside this utility function.  That mixes responsibilities.

Reply:
agreed, done